integration-cli/docker_cli_daemon_test.go

```diff
@@ -2174,27 +2174,33 @@ func (s *DockerDaemonSuite) TestDaemonStartWithoutColors(c *check.C) {
 
 	infoLog := "\x1b[34mINFO\x1b"
 
+	b := bytes.NewBuffer(nil)
+
 	p, tty, err := pty.Open()
 	c.Assert(err, checker.IsNil)
-	defer func() {
-		tty.Close()
-		p.Close()
-	}()
-
-	b := bytes.NewBuffer(nil)
 	go io.Copy(b, p)
```

Comment:
Not directly related to this fix but shouldn't we wait for this copy to finish. Something like:
```
ch := make(chan struct{})
go func() {
  io.Copy()
  close(ch)
}()
...
d.stop()
<-ch
```